tpot/tpot.py

```diff
@@ -724,41 +724,17 @@ def decision_tree(input_df, max_features, max_depth):
         if max_depth < 1:
             max_depth = None
 
-        # If there are no features left (i.e., only 'class', 'group', and 'guess' remain in the DF), then there is nothing to do
-        if len(input_df.columns) == 3:
-            return input_df
-        
-        input_df = input_df.copy()
-
-        training_features = input_df.loc[input_df['group'] == 'training'].drop(['class', 'group', 'guess'], axis=1).values
-        training_classes = input_df.loc[input_df['group'] == 'training', 'class'].values
-
-        dtc = DecisionTreeClassifier(max_features=max_features,
-                                     max_depth=max_depth,
-                                     random_state=42)
-
-        dtc.fit(training_features, training_classes)
-
-        all_features = input_df.drop(['class', 'group', 'guess'], axis=1).values
-        input_df.loc[:, 'guess'] = dtc.predict(all_features)
-
-        # Also store the guesses as a synthetic feature
-        sf_hash = '-'.join(sorted(input_df.columns.values))
-        sf_hash += 'DT-{}-{}'.format(max_features, max_depth)
-        sf_identifier = 'SyntheticFeature-{}'.format(hashlib.sha224(sf_hash.encode('UTF-8')).hexdigest())
-        input_df.loc[:, sf_identifier] = input_df['guess'].values
-
-        return input_df
+        return TPOT._train_model_and_predict(input_df, DecisionTreeClassifier, max_features=max_features, max_depth=max_depth, random_state=42)
```

Comment:
I think the `TPOT.` is extraneous: It should be possible to call `_train_model_and_predict()` without it.